app-libs/stf/Cargo.toml

```diff
@@ -39,6 +39,8 @@ sp-runtime = { default-features = false, git = "https://github.com/paritytech/su
 my-node-runtime = { package = "integritee-node-runtime", optional = true, git = "https://github.com/integritee-network/integritee-node.git", branch = "master" }
 pallet-parentchain = { default-features = false, git = "https://github.com/integritee-network/pallets.git", branch = "master" }
 
+# simplyR
+simplyr = {path = "simplyR", default-features = false}
```

Comment:
As mentioned before, you can't depend on `simpliR` here because it is not `no_std` compatible, you may only depend on `simplyr-lib`.

Also, I would personally recommend not to include a git submodule, but use git dependencies:

```
simplyr-lib = { default-features = false, git = "https://github.com/BESTenergytrade/simplyR.git", branch = "master" }
```

Reply:
I'll switch to git dependencies.
Another question: should I directly implement the `simplyr-lib` inside the getter block?

Reply:
I don't really understand the question here, but it seems OK the way you did it now, although concept wise you should just pay attention to:
* `TrustedGetter`: no state mutation should happen, any changes to the state will be discarded after the execution of the trusted call.
* `TrustedCall`: state mutation happens.

Reply:
Sorry for the confusion. 
I mean can I directly call the methods from `simplyr-lib` inside the trusted block?
For example:
```
TrustedGetter::pay_as_bid_algo(_who) => {
	// list of orders
	let order_1 = Order {
		id: 1,
		order_type: OrderType::Ask,
		time_slot: "2022-03-04T05:06:07+00:00".to_string(),
		actor_id: "actor_1".to_string(),
		cluster_index: Some(0),
		energy_kwh: 2.0,
		price_euro_per_kwh: 0.3,
	};

	let order_2 = Order {
		id: 2,
		order_type: OrderType::Bid,
		time_slot: "2022-03-04T05:06:07+00:00".to_string(),
		actor_id: "actor_2".to_string(),
		cluster_index: Some(0),
		energy_kwh: 1.5,
		price_euro_per_kwh: 0.35,
	};

	// create a market input
	let market_input = MarketInput { orders: vec![order_1, order_2] };

	let pay_as_bid: MarketOutput = pay_as_bid_matching(&market_input);
	Some(pay_as_bid.encode())
},
```

2. Can you please again take a look again if `simplyR` is not `no_std` compatible? I still see the same error.

Reply:
@clangenb May be you can just look at the last commit.

Reply:
1. Yes you can, you can call any pub function of a rust crate.
2. In this commit you broke `no_std` compatibility: https://github.com/BESTenergytrade/simplyR/commit/8ca040d542273cad6278be0e8160c72c7bb43548

you need to add the `default-features = false` to the `codec` dependencies

Reply:
Also, maybe you should check with Timo your branch on simplyR. I don't think you need to change the `simplyR` to be a library. It seems to me that you only need to import `simplyr-lib`, which serves everything you need.

Additionally, the two of you seem to use different formatters, make sure that both of you use rustfmt as the formatter, which is an IDE setting. To be bulletproof that you use the exact same formatting settings, and be independent of rust versions, you can use a [rustfmt.toml](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search).

Reply:
I still see the same error even after using the `default-features = false`. :(

Reply:
Did you run `cargo update -p simplyr-lib` to ensure that you have the newest commit of your git dependency?

Reply:
If you want to experiment, you can also patch git dependencies with local ones with this pattern:
https://github.com/integritee-network/worker/blob/543b3f78d0750f609f3c2a7c2986a1b6393af480/Cargo.toml#L92

Reply:
I used `cargo update` before.

Reply:
Still same issue even after `cargo update -p simplyr-lib`.

Reply:
This is the complete error:

```
error[E0152]: duplicate lang item in crate `std` (which `parity_scale_codec` depends on): `panic_impl`.
  |
  = note: the lang item is first defined in crate `sgx_tstd` (which `itp_enclave_metrics` depends on)
  = note: first definition in `sgx_tstd` loaded from /opt/shared/worker/enclave-runtime/target/release/deps/libsgx_tstd-19bc28e99ffd6263.rmeta
  = note: second definition in `std` loaded from /root/.rustup/toolchains/nightly-2022-10-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-7b2106000b625742.rlib

For more information about this error, try `rustc --explain E0152`.
error: could not compile `itp-enclave-metrics` due to previous error
warning: build failed, waiting for other jobs to finish...
error[E0152]: duplicate lang item in crate `sgx_tstd` (which `getrandom` depends on): `panic_impl`.
  |
  = note: the lang item is first defined in crate `std` (which `parity_scale_codec` depends on)
  = note: first definition in `std` loaded from /root/.rustup/toolchains/nightly-2022-10-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-7b2106000b625742.rlib
  = note: second definition in `sgx_tstd` loaded from /opt/shared/worker/enclave-runtime/target/release/deps/libsgx_tstd-19bc28e99ffd6263.rmeta

error: could not compile `sp-core` due to previous error
make[1]: *** [Makefile:56: libenclave.a] Error 101
make[1]: Leaving directory '/opt/shared/worker/enclave-runtime'
make: *** [Makefile:208: enclave] Error 2
```